tests/imodels-clients-tests/src/unit/management/AxiosRestClient.test.ts

```diff
@@ -0,0 +1,103 @@
+
+describe("[Management] AxiosRestClient", () => {
+  let axiosMock: MockAdapter;
+
+  before(() => {
+    axiosMock = new MockAdapter(axios);
+  });
+
+  it("should get response body", async () => {
+    // Arrange
+    const requestParams: HttpRequestWithJsonBodyParams = {
+      url: "https://imodelhub.bentley.com",
+      headers: {},
+      body: {
+        content: {},
+        contentType: ContentType.Json
+      }
+    };
+    const responseBody = {
+      iModelId: "IMODEL_ID",
+      changesetId: 4,
+      _links: {
+        self: "https://imodelhub.bentley.com/something"
+      }
+    };
+    axiosMock.onPost(requestParams.url).reply(200, responseBody, {});
+    const restClient = new AxiosRestClient(IModelsErrorParser.parse);
+
+    // Act
+    const response = await restClient.sendPostRequest(requestParams);
+
+    // Assert
+    expect(response.body).to.deep.equal(responseBody);
+  });
+
+  it("should get response header value by case-insensitive name", async () => {
+    // Arrange
+    const requestParams: HttpRequestWithJsonBodyParams = {
+      url: "https://imodelhub.bentley.com",
+      headers: {},
+      body: {
+        content: {},
+        contentType: ContentType.Json
+      }
+    };
+    const responseHeaders = {
+      "location": "https://some-url.com",
+      "retry-after": 60
+    };
+    axiosMock.onPost(requestParams.url).reply(200, {}, responseHeaders);
+    const restClient = new AxiosRestClient(IModelsErrorParser.parse);
+
+    // Act & Assert
+    const response = await restClient.sendPostRequest(requestParams);
+
+    expect(response.headers.get("location")).to.equal("https://some-url.com");
+    expect(response.headers.get("Location")).to.equal("https://some-url.com");
+    expect(response.headers.get("LOCATION")).to.equal("https://some-url.com");
+    expect(response.headers.get("Retry-After")).to.equal("60");
+    expect(response.headers.get("non-existent-header")).to.not.exist;
+  });
+
+  it("should get all response headers", async () => {
+    // Arrange
+    const requestParams: HttpRequestWithJsonBodyParams = {
+      url: "https://imodelhub.bentley.com",
+      headers: {},
+      body: {
+        content: {},
+        contentType: ContentType.Json
+      }
+    };
+    const responseHeaders = {
+      "location": "https://some-url.com",
+      "retry-after": 60,
+      "Content-Type": "application/json"
+    };
+    const expectedHeaders = {
+      "location": "https://some-url.com",
+      "retry-after": "60",
+      "Content-Type": "application/json"
+    };
+    axiosMock.onPost(requestParams.url).reply(200, {}, responseHeaders);
+    const restClient = new AxiosRestClient(IModelsErrorParser.parse);
+
+    // Act
+    const response = await restClient.sendPostRequest(requestParams);
+    const actualHeaders = response.headers.getAll();
+
+    // Assert
+    expect(JSON.stringify(actualHeaders)).to.deep.equal(JSON.stringify(expectedHeaders));
```

Comment:
do you really need to use JSON.stringify here? you're already using `deep.equal`, that should suffice

Reply:
It was complaining about incorrect type being returned since `actualHeaders` is derived class and contains more information that `expectedHeaders` has. Changed to Object assignment for comparison.